vars/xwikiBuild.groovy

```diff
@@ -179,6 +179,14 @@ void call(name = 'Default', body)
                     options: publishers)
             {
                 try {
+                    def branchName = env['BRANCH_NAME']
+                    // If we're building a feature branch (not master and not stable) we want to set a specific version
+                    // for it so that we don't mess up the snapshot repository.
+                    if (!isMasterBranch(branchName) && !isStableBranch(branchName)) {
+                        def branchVersion = "${pom.version}-${branchName}"
+                        echoXWiki "Setting version to: ${branchVersion}"
+                        sh script: "mvn -f ${pom} versions:set -DnewVersion=${branchVersion}"
```

Comment:
@vmassol 
> Ideally we should check the output result too and not do the next maven build if it failed

I used `sh script:` here, according to what's used few lines above it should stop the build if an error occurred during the command (see: https://github.com/xwiki/xwiki-jenkins-pipeline/blob/master/vars/xwikiBuild.groovy#L139)

Reply:
Indeed, sounds good (https://www.jenkins.io/doc/pipeline/steps/workflow-durable-task-step/#sh-shell-script).